src/ContentValidationListener.php

```diff
@@ -356,6 +367,101 @@ protected function useRawData($controllerService)
         return false;
     }
 
+    /**
+     * @param $controllerService
+     * @return bool
+     */
+    protected function shouldRemoveEmptyData($controllerService)
+    {
+        if (! isset($this->config[$controllerService]['remove_empty_data'])
+            || (isset($this->config[$controllerService]['remove_empty_data'])
+                && $this->config[$controllerService]['remove_empty_data'] === true)
+        ) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * @param array $data Data to filter null values from
+     * @param array $compareTo Original data, send along to preserve keys/values in $data which are intentional
+     *
+     * @return array
+     */
+    // TODO strict types for PHP > 7 when module gets bumped
+    //    protected function removeEmptyData(array $data, array $compareTo = []) : array
```

Comment:
Put this in the docblock, above the `@param` annotations, as an `@todo` annotation, please.

Reply:
done